### DISCUSSION TITLE
How Do Zed Developers Design Their Application UI? Are the using just rust? Or using something like XML to design their layout design template

### DISCUSSION THREAD

**Discussion by jaydeepkasture** [zed-industries/zed#13103] [Feature Requests]
Curious about the user interface design process behind the Zed text editor? Many development environments like Android Studio and Visual Studio use XML to design layouts, making it easier to create and manage complex interfaces. My question is: How do Zed developers manage to design their application's UI with Rust? Are they using a tool or framework similar to XML for layout design, or are they creating the entire layout using pure Rust code? Designing a complex UI with just code can be challenging, so I'm interested in learning more about their approach and the tools they use.

I am expecting what apporch did Zed developer used to design their application layout

**Comment by notpeter** [comment]
See [GPUI](https://www.gpui.rs/) A fast, productive UI framework for Rust from the creators of Zed.

Check out [Leveraging Rust and the GPU to render user interfaces at 120 FPS](https://zed.dev/blog/videogame) on the Zed blog for an introduction.

**Comment by SuperKenVery** [comment]
> Designing a complex UI with just code can be challenging

Well, they are using **declarative** GUI so it's not that difficult. Pretty much like how SwiftUI works.
